portal-impl/src/com/liferay/portal/service/impl/ResourcePermissionLocalServiceImpl.java

```diff
@@ -90,6 +90,36 @@
 	 */
 	public static final String[] EMPTY_ACTION_IDS = {null};
 
+	@Override
+	public void addResourcePermission(
+			long companyId, String name, int scope, long roleId,
+			String actionId, String[] groupIds)
+		throws Exception {
+
+		if (scope == ResourceConstants.SCOPE_COMPANY) {
+			addResourcePermission(
+				companyId, name, scope, String.valueOf(companyId), roleId,
+				actionId);
+		}
+		else if (scope == ResourceConstants.SCOPE_GROUP_TEMPLATE) {
+			addResourcePermission(
+				companyId, name, ResourceConstants.SCOPE_GROUP_TEMPLATE,
```

Comment:
It would be better to use the variable "scope" as the third parameter instead of the constant for consistency with the previous clause.